src/main.rs

```diff
@@ -130,8 +177,8 @@ enum Format {
     WithRelativeOffset(Expr, Box<Format>),
     /// Transform a decoded value with a function
     Map(Func, Box<Format>),
-    /// Conditional format
-    If(Expr, Box<Format>, Box<Format>),
+    /// Pattern match on an expression
+    Match(Expr, Vec<(Pattern, Format)>),
```

Comment:
Documentation could be better here… it was hard to document this (mentioning the the byte strings matched by the format), because this format is matching on an expression as well 😅